packages/grid/src/vaadin-grid-selection-mixin.js

```diff
@@ -77,12 +77,8 @@ export const SelectionMixin = (superClass) =>
     }
 
     /** @private */
-    _selectedItemsChanged(e) {
-      if (this.$.items.children.length && (e.path === 'selectedItems' || e.path === 'selectedItems.splices')) {
-        Array.from(this.$.items.children).forEach((row) => {
-          this._updateItem(row, row._item);
-        });
-      }
```

Comment:
The current selected items change observer calls `_updateItem` function manually for _all_ body row elements, even if they're in loading state, or completely out of use / hidden.